src/test/java/org/gridsuite/modification/server/utils/MatcherGeneratorModificationInfos.java

```diff
@@ -0,0 +1,90 @@
+/*
+  Copyright (c) 2021, RTE (http://www.rte-france.com)
+  This Source Code Form is subject to the terms of the Mozilla Public
+  License, v. 2.0. If a copy of the MPL was not distributed with this
+  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.gridsuite.modification.server.utils;
+
+import org.gridsuite.modification.server.dto.GeneratorModificationInfos;
+import org.gridsuite.modification.server.dto.ReactiveCapabilityCurveModificationInfos;
+import org.hamcrest.Description;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * @author Ayoub Labidi <ayoub.labidi at rte-france.com>
+ */
+public class MatcherGeneratorModificationInfos extends MatcherModificationInfos<GeneratorModificationInfos> {
+
+    protected MatcherGeneratorModificationInfos(GeneratorModificationInfos ref) {
+        super(ref);
+    }
+
+    public static MatcherGeneratorModificationInfos createMatcherGeneratorModificationInfos(GeneratorModificationInfos generatorModificationInfos) {
+        return new MatcherGeneratorModificationInfos(generatorModificationInfos);
+    }
+
+    @Override
+    public boolean matchesSafely(GeneratorModificationInfos m) {
+        return super.matchesSafely(m)
+                && m.getEquipmentId().equals(reference.getEquipmentId())
+                && m.getSubstationIds().equals(reference.getSubstationIds())
+                && m.getEquipmentName().equals(reference.getEquipmentName())
+                && Objects.equals(m.getEnergySource(), reference.getEnergySource())
```

Comment:
Missing (like in MatcherLoadCreationInfos):
&& m.getVoltageLevelId().equals(reference.getVoltageLevelId())
&& m.getBusOrBusbarSectionId().equals(reference.getBusOrBusbarSectionId())

from InjectionCreationInfos
Maybe we could have MatcherInjectionCreationInfos

Reply:
Done